libdebug/debugger/internal_debugger.py

```diff
@@ -993,30 +1026,104 @@ def get_thread_by_id(self: InternalDebugger, thread_id: int) -> ThreadContext:
+                backing_file = self._get_process_full_path()
+                liblog.debugger(
+                    f"No backing file specified and no correspondant absolute address for {hex(address)}. Assuming {backing_file}."
+                )
+        elif (
+            backing_file == (full_backing_path := self._get_process_full_path())
+            or backing_file == "binary"
+            or backing_file == self._get_process_name()
+        ):
+            backing_file = full_backing_path
+
+        filtered_maps = []
+        unique_files = set()
+
+        for vmap in maps:
+            if backing_file in vmap.backing_file:
+                filtered_maps.append(vmap)
+                unique_files.add(vmap.backing_file)
+
+        if len(unique_files) > 1:
+            raise ValueError(
+                f"The substring {backing_file} is present in multiple, different backing files. The address resolution cannot be accurate."
+            )
+
+        if not filtered_maps:
+            raise ValueError(f"The specified string {backing_file} does not correspond to any backing file.")
+        return normalize_and_validate_address(address, filtered_maps)
+
+    def resolve_symbol(self: InternalDebugger, symbol: str, backing_file: str) -> int:
         """Resolves the address of the specified symbol.
 
         Args:
             symbol (str): The symbol to resolve.
+            backing_file (str): The backing file to resolve the symbol in.
 
         Returns:
             int: The address of the symbol.
         """
         maps = self.debugging_interface.maps()
-        address = resolve_symbol_in_maps(symbol, maps)
-        return normalize_and_validate_address(address, maps)
+
+        if backing_file == "absolute":
+            raise ValueError("Cannot use `absolute` backing file with symbols.")
+
+        if backing_file == "default":
+            # If no explicit backing file is specified, we have to assume it is in the main map
+            backing_file = self._get_process_full_path()
+            liblog.debugger(f"No backing file specified for the symbol {symbol}. Assuming {backing_file}.")
+        elif (
+            backing_file == (full_backing_path := self._get_process_full_path())
+            or backing_file == "binary"
+            or backing_file == self._get_process_name()
+        ):
+            backing_file = full_backing_path
+
+        filtered_maps = []
+        unique_files = set()
+
+        for vmap in maps:
+            if backing_file in vmap.backing_file:
+                filtered_maps.append(vmap)
+                unique_files.add(vmap.backing_file)
+
+        if len(unique_files) > 1:
+            raise ValueError(
+                f"The substring {backing_file} is present in multiple, different backing files. The address resolution cannot be accurate."
+            )
+
+        if not filtered_maps:
+            raise ValueError(f"The specified string {backing_file} does not correspond to any backing file.")
```

Comment:
Maybe here too? Print the set of all backing files to choose from

Reply:
67a9609f8c813654d939a28ea5970f65eff64650